model/event/auth_event.go

```diff
@@ -0,0 +1,107 @@
+package event
+
+import (
+	"time"
+
+	"github.com/evergreen-ci/gimlet"
+	"github.com/pkg/errors"
+)
+
+const (
+	ResourceTypeUser = "USER"
+	ResourceTypeRole = "ROLE"
+)
+
+// UserEventType represents types of changes possible to a DB user.
+type UserEventType string
+
+const (
+	UserEventTypeRolesUpdate UserEventType = "USER_ROLES_UPDATED"
+)
+
+func (e UserEventType) validate() error {
+	switch e {
+	case UserEventTypeRolesUpdate:
+		return nil
+	default:
+		return errors.Errorf("invalid user event type '%s'", e)
+	}
+}
+
+type userData struct {
+	User   string      `bson:"user" json:"user"`
+	Before interface{} `bson:"before" json:"before"`
+	After  interface{} `bson:"after" json:"after"`
+}
+
+// LogUserEvent logs a DB User change to the event log collection.
+func LogUserEvent(user string, eventType UserEventType, before, after interface{}) error {
+	if err := eventType.validate(); err != nil {
+		return errors.Wrapf(err, "failed to log user event for user '%s'", user)
+	}
+
+	data := userData{
+		User:   user,
+		Before: before,
+		After:  after,
+	}
+	event := EventLogEntry{
+		Timestamp:    time.Now(),
+		EventType:    string(eventType),
+		Data:         data,
+		ResourceType: ResourceTypeUser,
+	}
+	logger := NewDBEventLogger(AllLogCollection)
+	if err := logger.LogEvent(&event); err != nil {
+		return errors.Wrapf(err, "failed to log user event for user '%s'", user)
+	}
+
+	return nil
+}
+
+// RoleEventType represents types of changes possible to a role.
+type RoleEventType string
+
+const (
+	RoleEventTypeAdd    RoleEventType = "ROLE_ADDED"
```

Comment:
I don't understand what sorts of changes these can be. The role events should be changes to a role, such as changing the permissions or name

Reply:
You don't want to log adding or deleting a role to the role collection?

Reply:
I think those are less important, since just creating a role doesn't grant anyone anything. It might be better to defer this until we figure out how mana will interact with evergreen, since that will primarily be the one making changes to roles

Reply:
deleted the code, but I think it will still be useful down the road because MANA will inevitably have to go through evergreen to make the changes so there should be some type of event logging.